java/src/main/java/com/google/api/services/samples/youtube/cmdline/live/GetLiveChatId.java

```diff
@@ -0,0 +1,134 @@
+        // authenticated user's account, but not other types of account access.
+        List<String> scopes = Lists.newArrayList(YouTubeScopes.YOUTUBE_READONLY);
+
+        try {
+            // Authorize the request.
+            Credential credential = Auth.authorize(scopes, "getlivechatid");
+
+            // This object is used to make YouTube Data API requests.
+            youtube = new YouTube.Builder(Auth.HTTP_TRANSPORT, Auth.JSON_FACTORY, credential)
+                .setApplicationName("youtube-cmdline-getlivechatid-sample").build();
+
+            // Get the liveChatId
+            String liveChatId = getLiveChatId(youtube, args.length == 1 ? args[0] : null);
+            if (liveChatId != null) {
+                System.out.println("Live chat id: " + liveChatId);
+            } else {
+                System.err.println("Unable to find a live chat id");
+                System.exit(1);
+            }
+        } catch (GoogleJsonResponseException e) {
+            System.err
+                .println("GoogleJsonResponseException code: " + e.getDetails().getCode() + " : "
+                    + e.getDetails().getMessage());
+            e.printStackTrace();
+
+        } catch (IOException e) {
+            System.err.println("IOException: " + e.getMessage());
+            e.printStackTrace();
+        } catch (Throwable t) {
+            System.err.println("Throwable: " + t.getMessage());
+            t.printStackTrace();
+        }
+    }
+
+    /**
+     * Retrieves the liveChatId from the videoId, if specified. If not specified, will find the
+     * liveChatId for the signed in user's live broadcast.
+     *
+     * @param youtube The object is used to make YouTube Data API requests.
+     * @param videoId The videoId associated with the live chat (optional).
+     * @return A liveChatId, or null if not found.
+     */
+    static String getLiveChatId(YouTube youtube, String videoId) throws IOException {
+        if (videoId != null && !videoId.isEmpty()) {
+            // Get liveChatId from the video
+            YouTube.Videos.List videoList = youtube.videos()
+                .list("liveStreamingDetails")
+                .setFields("items/liveStreamingDetails/activeLiveChatId")
+                .setId(videoId);
+            VideoListResponse response = videoList.execute();
+            for (Video v : response.getItems()) {
+                String liveChatId = v.getLiveStreamingDetails().getActiveLiveChatId();
+                if (liveChatId != null && !liveChatId.isEmpty()) {
+                    return liveChatId;
+                }
+            }
+        } else {
+            // Get signed in user's liveChatId
+            YouTube.LiveBroadcasts.List broadcastList = youtube
+                .liveBroadcasts()
+                .list("snippet")
+                .setFields("items/snippet/liveChatId")
+                .setBroadcastType("all")
+                .setBroadcastStatus("active");
```

Comment:
No, there will only be one liveChatId in both cases. Had another API dev code review it and made a few other minor changes.